gtda/externals/python/subsampling_interface.py

```diff
@@ -0,0 +1,110 @@
+    :type starting_point: unsigned.
+    :returns:  The subsample point set.
+    :rtype: vector[vector[double]]
+    """
+    if off_file:
+        if os.path.isfile(off_file):
+            if starting_point == '':
+                return subsampling_n_farthest_points_from_file(off_file.encode('utf-8'),
+                                                               nb_points)
+            else:
+                return subsampling_n_farthest_points_from_file(off_file.encode('utf-8'),
+                                                               nb_points,
+                                                               starting_point)
+        else:
+            print("file " + off_file + " not found.")
+    else:
+        if points is None:
+            # Empty points
+            points = []
+        if starting_point == '':
+            return subsampling_n_farthest_points(points, nb_points)
+        else:
+            return subsampling_n_farthest_points(points, nb_points,
+                                                 starting_point)
+
+
+def pick_n_random_points(points=None, off_file='', nb_points=0):
+    """Subsample a point set by picking random vertices.
+
+    :param points: The input point set.
+    :type points: vector[vector[double]].
+
+    Or
+
+    :param off_file: An OFF file style name.
+    :type off_file: string
+
+    :param nb_points: Number of points of the subsample.
+    :type nb_points: unsigned.
+    :returns:  The subsample point set.
+    :rtype: vector[vector[double]]
+    """
+    if off_file:
+        if os.path.isfile(off_file):
+            return subsampling_n_random_points_from_file(off_file.encode('utf-8'),
+                nb_points)
+        else:
+            print("file " + off_file + " not found.")
+    else:
+        if points is None:
+            # Empty points
+            points = []
+        return subsampling_n_random_points(points, nb_points)
+
+
+def sparsify_point_set(points=None, off_file='', min_squared_dist=0.0):
+    """Outputs a subset of the input points so that the squared distance
+    between any two points is greater than or equal to min_squared_dist.
+
+    :param points: The input point set.
+    :type points: vector[vector[double]].
+
+    Or
+
+    :param off_file: An OFF file style name.
+    :type off_file: string
+
+    :param min_squared_dist: Minimum squared distance separating the output \
+    points.
+    :type min_squared_dist: float.
+    :returns:  The subsample point set.
+    :rtype: vector[vector[double]]
+    """
+    if off_file:
+        if os.path.isfile(off_file):
+            return subsampling_sparsify_points_from_file(off_file.encode('utf-8'),
+                                                         min_squared_dist)
+        else:
+            print("file " + off_file + " not found.")
+    else:
+        if points is None:
+            # Empty points
+            points = []
+        return subsampling_sparsify_points(points, min_squared_dist)
```

Comment:
I would add a header to the file explaining why this is unused in giotto-tda.
I'm not even sure dead code should be pushed

Reply:
If those are the bindings for a Transformer to appear in a future release, I think it is fine. But I am not sure what is the situation with CGAL right now?